src/sail_json_backend/json.ml

```diff
@@ -614,6 +614,8 @@ let json_of_syntax k =
         if String.equal s "sep" then ","
         else if String.equal s "\"(\"" then "("
         else if String.equal s "\")\"" then ")"
+        else if String.starts_with ~prefix:"maybe_" s then "[," ^ remove_identity_funcs s ^ "]"
+        else if String.contains s ',' then List.hd (Str.split (Str.regexp ",") (remove_identity_funcs s))
```

Comment:
Since we've done a similar look up with constants before (the fix to the order of operands,) I think it makes sense for me to work on it since I already know the approach.